File: n3/3.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        Console.Write("Введите количество чисел и желаемые числа: ");
        string[] input = Console.ReadLine().Split();
        int amount = int.Parse(input[0]); // количество чисел
        string[] numbers = new string[amount];
        Array.Copy(input, 1, numbers, 0, amount);

        List<int> odd = new List<int>(); // список нечётных цифр
        int count = 0; // счётчик нечётных цифр

        // перебор чисел
        foreach (string num in numbers) {
            foreach (char digit in num) {
                if ((digit - '0') % 2 == 1) { // проверка на нечётность
                    count++;
                }
            }
            odd.Add(count); // добавление количества нечётных цифр
            count = 0; // сброс счётчика
        }
        Console.Write("Количество нечётных цифр в числах: ");
        foreach (int i in odd) {
            Console.Write(i + " ");
        }
        Console.WriteLine();
    }
}

File: n3/3.py
lst = list(map(str,input().split()))
amount = lst[1] # кол-во чисел
numbers = lst[1:] # список перебираемых чисел
odd = [] # список нечетных цифр
count = 0 # счетчик чисел
for i in numbers: 
    for i1 in range(len(i)):
        if int(i[i1]) % 2 == 1: # проверка на нечетность
            count += 1
    odd.append(count) # добавление кол-ва нечетных цифр в числе в список 
    count = 0  # обнуление счетчика дл перехода к следующему числу
print(odd)
